modules/beta-private-cluster/cluster.tf

```diff
@@ -501,8 +501,22 @@ resource "google_container_node_pool" "pools" {
   }
 
   upgrade_settings {
-    max_surge       = lookup(each.value, "max_surge", 1)
-    max_unavailable = lookup(each.value, "max_unavailable", 0)
+    strategy        = lookup(each.value, "strategy", "SURGE")
+    max_surge       = lookup(each.value, "strategy", "SURGE") == "SURGE" ? lookup(each.value, "max_surge", 1) : null
+    max_unavailable = lookup(each.value, "strategy", "SURGE") == "SURGE" ? lookup(each.value, "max_unavailable", 0) : null
+
+    dynamic "blue_green_settings" {
+      for_each = lookup(each.value, "strategy", "SURGE") == "BLUE_GREEN" ? [1] : []
+      content {
+        node_pool_soak_duration = lookup(each.value, "node_pool_soak_duration", "3600s")
```

Comment:
Could we default these to null too to use provider defaults?

Reply:
Done. :heavy_check_mark: